pilot/pkg/proxy/envoy/v2/ads_common.go

```diff
@@ -47,6 +115,10 @@ Loop:
 		}
 	}
 
+	if appliesToProxy {
```

Comment:
it's not that. ConfigsUpdated and namespacesUpdated are orthogonal and means only both them affect the proxy can we say the proxy needs push rather than returning true when the namespace check give a "true" result like the code beblow.